packages/mdc-select/index.ts

```diff
@@ -0,0 +1,596 @@
+    if (this.menu_) {
+      this.menu_.unlisten(menuSurfaceConstants.strings.CLOSED_EVENT, this.handleMenuClosed_);
+      this.menu_.unlisten(menuSurfaceConstants.strings.OPENED_EVENT, this.handleMenuOpened_);
+      this.menu_.unlisten(menuConstants.strings.SELECTED_EVENT, this.handleMenuSelected_);
+      this.menu_.destroy();
+    }
+
+    if (this.ripple) {
+      this.ripple.destroy();
+    }
+    if (this.outline_) {
+      this.outline_.destroy();
+    }
+    if (this.leadingIcon_) {
+      this.leadingIcon_.destroy();
+    }
+    if (this.helperText_) {
+      this.helperText_.destroy();
+    }
+    if (this.validationObserver_) {
+      this.validationObserver_.disconnect();
+    }
+
+    super.destroy();
+  }
+
+  getDefaultFoundation(): MDCSelectFoundation {
+    return new MDCSelectFoundation({
+          ...(this.nativeControl_ ? this.getNativeSelectAdapterMethods_() : this.getEnhancedSelectAdapterMethods_()),
+          ...this.getCommonAdapterMethods_(),
+          ...this.getOutlineAdapterMethods_(),
+          ...this.getLabelAdapterMethods_(),
+        },
+        this.getFoundationMap_(),
+    );
+  }
+
+  /**
+   * Handles setup for the enhanced menu.
+   */
+  private enhancedSelectSetup_(menuFactory: MenuFactory) {
+    const isDisabled = this.root_.classList.contains(cssClasses.DISABLED);
+    this.selectedText_!.setAttribute('tabindex', isDisabled ? '-1' : '0');
+    this.hiddenInput_ = this.root_.querySelector(strings.HIDDEN_INPUT_SELECTOR);
+    this.menuElement_ = this.root_.querySelector(strings.MENU_SELECTOR)!;
+    this.menu_ = menuFactory(this.menuElement_);
+    this.menu_.hoistMenuToBody();
+    this.menu_.setAnchorElement(this.root_);
+    this.menu_.setAnchorCorner(menuSurfaceConstants.Corner.BOTTOM_START);
+    this.menu_.wrapFocus = false;
+  }
+
+  private initRipple_(): MDCRipple {
+    // At least one of the elements must be non-null.
+    const element = this.nativeControl_ || this.selectedText_!;
+
+    // tslint:disable:object-literal-sort-keys
+    const foundation = new MDCRippleFoundation({
+      ...MDCRipple.createAdapter(this),
+      registerInteractionHandler: <E extends EventType>(evtType: E, handler: SpecificEventListener<E>) => {
+        element.addEventListener(evtType, handler);
+      },
+      deregisterInteractionHandler: <E extends EventType>(evtType: E, handler: SpecificEventListener<E>) => {
+        element.removeEventListener(evtType, handler);
+      },
+    });
+    // tslint:enable:object-literal-sort-keys
+
+    return new MDCRipple(this.root_, foundation);
+  }
+
+  private getNativeSelectAdapterMethods_() {
+    // tslint:disable:object-literal-sort-keys
+    return {
+      getValue: () => this.nativeControl_!.value,
+      setValue: (value: string) => this.nativeControl_!.value = value,
+      openMenu: () => undefined,
+      closeMenu: () => undefined,
+      isMenuOpen: () => false,
+      setSelectedIndex: (index: number) => this.nativeControl_!.selectedIndex = index,
```

Comment:
More adapter methods that are supposed to have no return value

Reply:
Done